pkg/controller/apiextensions/claim/reconciler.go

```diff
@@ -303,8 +303,7 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 					// after a brief wait, in case this was a transient error.
 					log.Debug("Cannot remove finalizer", "error", err, "requeue-after", time.Now().Add(aShortWait))
 					record.Event(cm, event.Warning(reasonDelete, err))
-					cm.SetConditions(v1alpha1.Deleting(), v1alpha1.ReconcileError(err))
```

Comment:
I noticed there are a few places where you removed setting the `Deleting` condition also, but didn't notice that explicitly called out in the PR or the issue (may have missed it). Do you think we the event / log in deleting scenarios should replace setting this condition?

Reply:
Good catch - I forgot to mention this. It's another case where it probably doesn't make sense to have N controllers (composite, claim, RBAC, etc) all trying to set the one condition - in this case `Deleting()` tries to set `Established: False` in both XRD controllers.

We could address this by using two different conditions - perhaps `Terminating()` (like CRD) for the definition controller that's cleaning up the composite's CRD, and `Redacting()` for the offered controller that's cleaning up the claim's CRD. I opted to take the lazy way out and just not set a condition when we're being deleted, because we're already emitting an event and because in the happy path the XRD won't stick around very long after being deleted. Do you think there's value in keeping these "deleting" conditions?

Reply:
I like the two different conditions here I think, but am kind of indifferent on whether they need to be part of this PR.